### ISSUE TITLE
incompatible with pending Nix 2.10 changes

### ISSUE THREAD

**Issue by nrdxp** [divnix/std#77]
Running with nixos/nix#6530, which is currently planned for the next point release, I am getting the following eval error pretty much everywhere in a work project:
```
error: evaluation aborted with the following error message: '

       Everything under ${cellsFrom}/* is considered a Cell

       Cells are directories by convention and therefore
       only directories are allowed at ${cellsFrom}/*

       Please remove '${cellsFrom}/std' and don't forget to add the change to version control.
       '
```

Not sure the root cause yet. Will post back when I know more

**Comment by nrdxp** [comment]
okay, so with the follow diff:
```diff
diff --git a/src/grow.nix b/src/grow.nix
index 1116a3b..a4a5726 100644
--- a/src/grow.nix
+++ b/src/grow.nix
@@ -86,7 +86,7 @@
     # Validations ...
     Organelles = validate.Organelles organelles;
     Systems = validate.Systems systems;
-    Cells = l.mapAttrsToList (validate.Cell cellsFrom Organelles) (l.readDir cellsFrom);
+    Cells = l.mapAttrsToList (validate.Cell cellsFrom Organelles) (l.traceVal (l.readDir cellsFrom));
 
     # Helpers ...
     accumulate =

```

It seems the type of `cells/std` is no longer known at this stage:
```
trace: { std = "unknown"; }
```

**Comment by blaggacao** [comment]
@nrdxp Is this issue still warranted or shuld we revive mitigation efforts in any way?

**Comment by nrdxp** [comment]
Well the lazy-trees branch still hasn't been merged: NixOS/nix#6530

I guess we'll have to wait and see what this looks like afterwards, as it is a pretty fast moving target.

**Comment by blaggacao** [comment]
- Is this still true?

- What can we do ahead of time?

**Comment by nrdxp** [comment]
* Last I checked yes. The lazy-trees branch seems to break a lot of flakes for me, to the point I haven't been able to test it much
* We can try to work out the breakage, but the lazy-tree branch still has changes coming in, so perhaps we should wait til it is at least out of draft

**Comment by blaggacao** [comment]
> changes coming in

changes yes, but maybe not breakages, any more? I'd like the idea to start thinking in forward-compatibility, here.

**Comment by nrdxp** [comment]
Yeah but as I said, it breaks a lot of stuff so it's kinda hard to test and pretty intimidating to start to fix. Feel free anytime though :sweat_smile:

**Comment by nrdxp** [comment]
Looks like the lazy-trees branch is finally out of draft, so it may be just the time to give it another shot.

**Comment by blaggacao** [comment]
These nix changes are pending and pending :hourglass: :smile:
